framework/src/postprocessors/SideAdvectiveFluxIntegral.C

```diff
@@ -0,0 +1,140 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SideAdvectiveFluxIntegral.h"
+#include "MathFVUtils.h"
+
+#include "metaphysicl/raw_type.h"
+
+registerMooseObject("MooseApp", SideAdvectiveFluxIntegral);
+registerMooseObject("MooseApp", ADSideAdvectiveFluxIntegral);
+
+template <bool is_ad, typename T>
+InputParameters
+SideAdvectiveFluxIntegralTempl<is_ad, T>::validParams()
+{
+  InputParameters params = SideIntegralPostprocessor::validParams();
+  MooseEnum component("x y z normal");
+
+  params.addRequiredParam<MooseEnum>("component", component, "The desired component of flux.");
+  params.addRequiredParam<MooseFunctorName>("vel_x", "x-component of the velocity vector");
+  params.addParam<MooseFunctorName>("vel_y", "y-component of the velocity vector");
+  params.addParam<MooseFunctorName>("vel_z", "z-component of the velocity vector");
+  params.addCoupledVar(
+      "advected_variable",
+      0,
+      "The advected variable quantity of which to compute advection flux; useful for "
+      "finite element simulations");
+  params.addParam<MaterialPropertyName>(
+      "advected_mat_prop",
+      0,
+      "The advected material property of which to compute advection flux; "
+      "useful for finite element simulations");
+  params.addParam<MooseFunctorName>("advected_quantity",
+                                    "The quantity to advect. This is the canonical parameter to "
+                                    "set the advected quantity when finite volume is being used.");
+
+  params.addClassDescription("Computes the volumetric advected quantity through a sideset.");
+
+  return params;
+}
+
+template <bool is_ad, typename T>
+SideAdvectiveFluxIntegralTempl<is_ad, T>::SideAdvectiveFluxIntegralTempl(
+    const InputParameters & parameters)
+  : SideIntegralPostprocessor(parameters),
+    _use_normal(getParam<MooseEnum>("component") == "normal"),
+    _component(getParam<MooseEnum>("component")),
+    _advected_variable_supplied(parameters.isParamSetByUser("advected_variable")),
+    _advected_variable(coupledValue("advected_variable")),
```

Comment:
use the ? : pattern with no default

Reply:
Is there a reason to better use `?:pattern` than default value?  In Coupleable.md, it seems to recommend use default value format.

Reply:
the default is seen by the user (as 0) in the documentation, when really it is ignored

the issue it also that when both are not specified, you have a non-zero flux computation, when you showed a default of 0 to the user so they expect 0

Reply:
This makes sense. I'll remove the default value. Thanks!